sympy/core/relational.py

```diff
@@ -1428,7 +1442,7 @@ def split_real_imag(expr):
                         rv = fuzzy_bool(Eq(*args))
                         if rv is True:
                             rv = None
-        elif any(a.is_infinite for a in Add.make_args(n)):
+        elif any(AssumptionsWrapper(a, assumptions).is_infinite for a in Add.make_args(n)):
```

Comment:
A one shot function would be better for cases like this. We don't need to make a wrapper instance just to call a single method:
```
def is_infinite(obj, assumptions):
    if assumptions is True:
         return obj.is_infinite
    else:
         return ask(Q.infinite(obj), assumptions)
```
I expect that to be faster and also to give tidier code when used.